puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/releaseTarballs.groovy

```diff
@@ -0,0 +1,61 @@
+pipeline {
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    stages {
+        parallel {
+            stage('Foreman') {
+                steps {
+                    verify_tag('foreman')
+                    build_tarball('foreman')
```

Comment:
The logic in these two jobs appears to be silly simple, enough that instead of continuing the generic job pattern, I think we are better off just creating functions for them and dropping those jobs. Just looking at `verify_tag`:

```
#!/bin/bash -xe

if [ "$(git tag -l ${version} | wc -l)" -ne 1 ]; then
  echo "ERROR: cannot find tag ${version}"
  git tag -l
  exit 1
fi

exit 0
```

That makes more sense to me as a function, gets rid of a job and makes this job more self contained. The same is likely true for `build_tarball`. Given this is a re-write of the `release_tarballs` job anyway, might as well take it the small step forward and avoid the code that triggers another job.

Reply:
:+1: on that. It also helps with the overview to have one pipeline rather than many subjobs.

Reply:
@ehelms from conversation earlier this week, I thought it was agreed upon that for now jobs would just be converted 1:1 rather than conversions along with refactoring.

Reply:
For moving jobs into source control a 1:1 conversion is what I was intending since its a net no change. If we are going to re-factor the job (which converting things to a pipeline is) then that's not 1:1 and we might as well include reasonable updates to fit the new style since new code is being written. If this is not exactly clear, we can clarify it more when we sync up.